src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useEffect, useMemo } from "react";
+import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useAsyncFn } from "react-use";
+import firebase from "firebase/app";
+
+import { MAX_TABLE_CAPACITY, MIN_TABLE_CAPACITY } from "settings";
+
+import { Table } from "types/Table";
+
+import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
+
+import { InputField } from "components/atoms/InputField";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableForm {
+  title: string;
+  subtitle?: string;
+  capacity: number;
+}
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  subtitle?: string;
+  capacity: number;
+  venueName: string;
+  tables: Table[];
+  tableOfUser?: Table;
+  onHide: () => void;
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  venueName,
+  title,
+  tables,
+  tableOfUser,
+  subtitle,
+  capacity,
+  onHide,
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+
+  const formDefaultValues = useMemo(
+    () => ({
+      title,
+      subtitle,
+      capacity,
+    }),
+    [capacity, subtitle, title]
+  );
+
+  const {
+    register,
+    handleSubmit,
+    errors,
+    watch,
+    reset,
+  } = useForm<EditTableForm>({
+    mode: "onBlur",
+    reValidateMode: "onBlur",
+    defaultValues: formDefaultValues,
+  });
+
+  // Updates the default values whenever they change. This is required because they are cached after the first render.
+  useEffect(() => {
+    reset(formDefaultValues);
+  }, [formDefaultValues, reset, title]);
+
+  // use useAsyncFn for easier error handling, instead of state hook
+  const [{ error }, updateTables] = useAsyncFn(
+    async (values: EditTableForm) => {
+      if (!user || !tableOfUser || !venueId) return;
+
+      const updatedTable = {
+        ...tableOfUser,
+        title: values.title,
+        subtitle: values.subtitle,
+        capacity: values.capacity,
+      };
+
+      return await firebase.functions().httpsCallable("venue-updateTables")({
+        venueId,
+        tables,
+        updatedTable,
+      });
+    },
+    [tableOfUser, tables, user, venueId]
+  );
+
+  const values = watch();
+
+  const onSubmit = useCallback(async () => {
+    await updateTables(values).then(() => {
+      onHide();
+    });
+  }, [onHide, updateTables, values]);
+
+  return (
+    <Modal show={isShown} onHide={onHide}>
+      <Modal.Body>
+        <form onSubmit={handleSubmit(onSubmit)} className="EditTableTitleModal">
+          <InputField
+            ref={register({ required: true })}
+            name="title"
+            defaultValue={title}
```

Comment:
Since you're setting the `defaultValue` in the `useForm` above, I don;t believe you also need to set it here. I believe it's generally more of a 'one or the other' situation.

See my other comments above in this review, or the docs for `defaultValues`, for more details:

- https://react-hook-form.com/api/useform
  - > You can set an `input's` default value with `defaultValue`/`defaultChecked` (read more from the React doc for Default Values). You can pass `defaultValues` as an optional argument to `useForm()` to populate the default values for the entire form, **or set values on an individual Controller component via its `defaultValue` property.**

Reply:
Whichever way you pick, you should be consistent with it (eg. the `subtitle` `InputField` below has no `defaultValue` prop on it; etc)

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1377/commits/c222264f9875e83b03a9a64b49c9ae1d57e9401c